teckers/teckers/AppDelegate.swift

```diff
@@ -7,18 +7,43 @@
 //
 
 import UIKit
+import Firebase
+import GoogleSignIn
 
 @UIApplicationMain
-class AppDelegate: UIResponder, UIApplicationDelegate {
+class AppDelegate: UIResponder, UIApplicationDelegate, GIDSignInDelegate {
 
     var window: UIWindow?
 
-
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
-        // Override point for customization after application launch.
+        // Use Firebase library to configure APIs
+        FirebaseApp.configure()
+        
+        GIDSignIn.sharedInstance().clientID = FirebaseApp.app()?.options.clientID
+        GIDSignIn.sharedInstance().delegate = self
+        
         return true
     }
-
+    
+    func sign(_ signIn: GIDSignIn!, didSignInFor user: GIDGoogleUser!, withError error: Error?) {
```

Comment:
Login a user is not a responsibility of the AppDelegate class, please move this logic to a different class

Reply:
I removed this

Reply:
I changed this function to view controller but I don't know if it's right

Reply:
I feel it should be better to have an specific class to handle Google Sign in & Firebase interaction and to call methods os that class from a controller or the app delegate if necessary